lib/forklift/box_distributor.rb

```diff
@@ -0,0 +1,163 @@
+module Forklift
+  class BoxDistributor
+
+    VAGRANTFILE_API_VERSION = '2'.freeze
+
+    def initialize(boxes)
+      @ansible_groups = {}
+      @boxes = boxes
+      @boxes = @boxes.keys.sort.each_with_object do |hash, key|
+        hash[key] = @boxes[key]
+        hash
+      end
+    end
+
+    def distribute
+      Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+        configure_vagrant_hostmanager(config)
+        configure_vagrant_cachier(config)
+
+        @boxes.each do |_name, box|
+          define_vm config, box
+        end
+
+        config.vm.provider :libvirt do |domain|
+          domain.memory = 4608
+          domain.cpus   = 2
+        end
+
+        config.vm.provider :virtualbox do |domain|
+          domain.memory = 4608
+          domain.cpus   = 2
+          domain.customize ['modifyvm', :id, '--natdnsproxy1', 'off']
+          domain.customize ['modifyvm', :id, '--natdnshostresolver1', 'off']
+        end
+      end
+    end
+
+    def define_vm(config, box = {})
+      config.vm.define box.fetch('name'), primary: box.fetch('default', false) do |machine|
+        machine.vm.box = box.fetch('box_name')
+        machine.vm.box_check_update = true
+        config.ssh.insert_key = false
+        machine.vm.box_url = box.fetch('box_url') if box.key?('box_url')
+
+        machine.vm.hostname = if box.fetch('hostname', false)
+                                box.fetch('hostname')
+                              else
+                                "#{box.fetch('name').to_s.tr('.', '-')}.example.com"
+                              end
+
+        networks = configure_networks(box.fetch('networks', []))
+        configure_shell(machine, box)
+        configure_ansible(machine, box['ansible'], box['name'])
+        configure_libvirt(machie, box, networks)
+        configure_virtualbox(machine, box)
+        configure_rackspace(machine, box)
+
+        yield machine if block_given?
+      end
+    end
+
+    def configure_vagrant_hostmanager(config)
+      return unless Vagrant.has_plugin?('vagrant-hostmanager')
+      config.hostmanager.enabled = true
+      config.hostmanager.manage_host = true
+      config.hostmanager.manage_guest = true
+      config.hostmanager.include_offline = true
+    end
+
+    def configure_vagrant_cachier(config)
+      return unless Vagrant.has_plugin?('vagrant-cachier')
+      config.cache.scope = :box
+      config.cache.synced_folder_opts = {
+        type: :nfs,
+        mount_options: ['rw', 'vers=3', 'tcp', 'nolock']
```

Comment:
Could we have this configurable? Last time I tried cacher I used nfs4 since that's what I use.